pkg/frontend/v2/frontend_test.go

```diff
@@ -98,17 +102,20 @@ func setupFrontendWithConcurrencyAndServerOptions(t *testing.T, reg prometheus.R
 	return f, ms
 }
 
-func sendResponseWithDelay(f *Frontend, delay time.Duration, userID string, queryID uint64, resp *httpgrpc.HTTPResponse) {
+func sendResponseWithDelay(t *testing.T, f *Frontend, delay time.Duration, userID string, queryID uint64, resp *httpgrpc.HTTPResponse) {
+	t.Helper()
+
 	if delay > 0 {
 		time.Sleep(delay)
 	}
 
 	ctx := user.InjectOrgID(context.Background(), userID)
-	_, _ = f.QueryResult(ctx, &frontendv2pb.QueryResultRequest{
+	_, err := f.QueryResult(ctx, &frontendv2pb.QueryResultRequest{
```

Comment:
`QueryResult` never returns `non-nil` error, so all changes to `sendResponseWithDelay` are unnecessary, and unrelated to this PR.